### PR TITLE
byData Analytics Adapter: add new analytics adapter

### PR DESCRIPTION
**Pull request by Prebid-bydata**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [x] Other

## Description of change

   adding a analytics adapter name byDataAnalyticsAdapter. 

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer - engineering@ascendeum.com
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/byDataAnalyticsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+          return {
+            name: data[i].name,
+            version: parseFloat(version)
+          };
+        }
+      }
+      return { name: 'unknown', version: 0 };
+    }
+  };
+  function generateUid() {
```
Comment (aleksatr) [inline modules/byDataAnalyticsAdapter.js:152]: @Prebid-bydata , do you have to generate uid exactly like this, would `utils.generateUUID()` work for you?

File: modules/byDataAnalyticsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+      { name: 'Internet Explorer', value: 'MSIE', version: 'MSIE' },
+      { name: 'Opera', value: 'Opera', version: 'Opera' },
+      { name: 'BlackBerry', value: 'CLDC', version: 'CLDC' },
+      { name: 'Mozilla', value: 'Mozilla', version: 'Mozilla' }
+    ],
+    init: function () {
+      var agent = this.header.join(' '); var os = this.matchItem(agent, this.dataos); var browser = this.matchItem(agent, this.databrowser);
+      return { os: os, browser: browser };
+    },
+    matchItem: function (string, data) {
```
Comment (aleksatr) [inline modules/byDataAnalyticsAdapter.js:121]: @Prebid-bydata , wouldn't it be better to do the whole user-agent parsing on the server?

File: modules/byDataAnalyticsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    ua['client_id'] = clientId;
+    ua['plateform_name'] = e.os.name;
+    ua['os_version'] = e.os.version;
+    ua['browser_name'] = e.browser.name;
+    ua['browser_version'] = e.browser.version;
+    ua['screen_size'] = screenSize;
+    ua['device_type'] = deviceType;
+    ua['time_zone'] = window.Intl.DateTimeFormat().resolvedOptions().timeZone;
+  }
+  var signedToken = getJWToken(ua);
```
Comment (aleksatr) [inline modules/byDataAnalyticsAdapter.js:208]: @Prebid-bydata what is the reasoning behind using JWT? (especially that secret key is not really secret)

File: modules/byDataAnalyticsAdapter.js [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+  let strJSON = JSON.stringify(obj);
+  logInfo(' sendPayload ', JSON.stringify(obj));
+  ajax(DEFAULT_EVENT_URL, undefined, strJSON, {
+    contentType: 'application/vnd.kafka.json.v2+json',
+    method: 'POST',
+    withCredentials: true
+  });
+}
+
+function addPbjsAndWinningsBids(isPbjs = true) {
```
[the remark this answers is not in the document]
Comment (Prebid-bydata) [reply modules/byDataAnalyticsAdapter.js:278]: hi @aleksatr , sure will update this change. thanks.

File: modules/byDataAnalyticsAdapter.js (hunk 1 above)
Comment (Prebid-bydata) [reply modules/byDataAnalyticsAdapter.js:152]: @aleksatr hi, our backend table uid setup like this, so we have to use in this format.

File: modules/byDataAnalyticsAdapter.js (hunk 3 above)
Comment (Prebid-bydata) [reply modules/byDataAnalyticsAdapter.js:205]: hi @aleksatr , we will use this in future , currently backend requirement.

File: modules/byDataAnalyticsAdapter.js (hunk 2 above)
Comment (Prebid-bydata) [reply modules/byDataAnalyticsAdapter.js:117]: hi @aleksatr thinking on this,  it tales micro seconds to execute.

**Comment by Prebid-bydata** [comment]
hello any update please.

**Comment by Prebid-bydata** [comment]
hi @aleksatr anything we need to do here , please update.

**Comment by ChrisHuie** [comment]
@Prebid-bydata can you add documentation for this new analytics adapter in our docs repo? (https://github.com/prebid/prebid.github.io)

Thanks 🙌

**Comment by Prebid-bydata** [comment]
hi @ChrisHuie documentation added for new analytics adapter and create PR (https://github.com/prebid/prebid.github.io/pull/3235) thanks.

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/3235

**Pull request merged**